extension/observer/ecsobserver/fetcher.go

```diff
@@ -86,24 +88,42 @@ func newTaskFetcher(opts taskFetcherOptions) (*taskFetcher, error) {
 		return nil, err
 	}
 
+	logger := opts.Logger
 	fetcher := taskFetcher{
-		logger:       opts.Logger,
-		ecs:          opts.ecsOverride,
-		ec2:          opts.ec2Override,
-		cluster:      opts.Cluster,
-		taskDefCache: taskDefCache,
-		ec2Cache:     ec2Cache,
-		// TODO: after the service matcher PR is merged, use actual service name filter here.
-		// For now, describe all the services
-		serviceNameFilter: func(name string) bool {
+		logger:            logger,
+		ecs:               opts.ecsOverride,
+		ec2:               opts.ec2Override,
+		cluster:           opts.Cluster,
+		taskDefCache:      taskDefCache,
+		ec2Cache:          ec2Cache,
+		serviceNameFilter: opts.serviceNameFilter,
+	}
+	// Match all the services for test. For production if there is no service related config,
+	// we don't describe any service, see serviceConfigsToFilter for detail.
+	if fetcher.serviceNameFilter == nil {
+		fetcher.serviceNameFilter = func(name string) bool {
 			return true
-		},
+		}
 	}
```

Comment:
Also, can we override `fetcher.serviceNameFilter` with this aways returning `true` function in unit test code?

Reply:
Yeah we can, I was too lazy to update all the test usages, I will create a factory func for tests.

Reply:
Updated in https://github.com/pingleig/opentelemetry-collector-contrib/commit/75c4a9c2becb8baf93ae2b03c94aed944585e56b